lib/logstash/inputs/redis.rb

```diff
@@ -165,21 +167,11 @@ def connect
   # private
   def load_batch_script(redis)
     #A Redis Lua EVAL script to fetch a count of keys
-    #in case count is bigger than current items in queue whole queue will be returned without extra nil values
-    redis_script = <<EOF
-          local i = tonumber(ARGV[1])
-          local res = {}
-          local length = redis.call('llen',KEYS[1])
-          if length < i then i = length end
-          while (i > 0) do
-            local item = redis.call("lpop", KEYS[1])
-            if (not item) then
-              break
-            end
-            table.insert(res, item)
-            i = i-1
-          end
-          return res
+      redis_script = <<EOF
+        local batchsize = tonumber(ARGV[1])
+        local result = redis.call('lrange', KEYS[1], 0, batchsize)
```

Comment:
My preference is to "block until there's data" Instead of doing a slow spin loop, but I won't block the PR on this preference.